addons/website/static/src/snippets/s_dynamic_snippet_carousel/dynamic_snippet_carousel.js

```diff
@@ -5,14 +5,167 @@ import { utils as uiUtils } from "@web/core/ui/ui_service";
+     */
+    getScrollMode() {
+        return this.el.classList.contains("o_carousel_multi_items") ? "single" : "all";
+    }
+
+    /**
+     * Show/hide the "Previous" arrow.
+     *
+     * Prevents users from navigating back from the first slide,
+     * since lazy loading only adds slides when moving forward.
+     */
+    togglePrevButton() {
+        const prev = this.el.querySelector(".carousel-control-prev");
+        if (!this.carouselEl || !prev) {
+            return;
+        }
+        const index = this.getActiveSlideIndex();
+        if (this.getScrollMode() === "single") {
+            prev.classList.toggle("d-none", this.hasMore && this.index == -1);
+        } else {
+            prev.classList.toggle("d-none", this.hasMore && index === 0);
+        }
+    }
+
+    /**
+     * Called on slide event. Detects if last slide is reached and
+     * load more data if available.
+     */
+    async onSlide(ev) {
+        if (ev.direction === "left") {
+            this.index++;
+        } else {
+            this.index--;
+        }
+        const slides = this.carouselEl.querySelectorAll(".carousel-item");
+        const index = this.getActiveSlideIndex();
+        if (
+            (this.hasMore && index === slides.length - 1) ||
+            (this.getScrollMode() === "single" &&
+                this.hasMore &&
+                this.index == this.data.length - 4)
+        ) {
+            ev.preventDefault();
+            await this.fetchData();
+            const carouselInstance = window.Carousel.getOrCreateInstance(this.carouselEl);
+            carouselInstance.next();
+        }
+    }
+
+    /**
+     * @override
+     */
+    getRpcParameters() {
+        return {
+            ...super.getRpcParameters(),
+            limit: this.limit,
+            offset: this.offset,
+        };
+    }
+
+    /**
+     * @override
+     */
+    renderContent() {
+        super.renderContent();
+        this.carouselEl = this.el.querySelector(".carousel");
+        this.index = -1;
+        this.togglePrevButton();
+    }
+
+    /**
+     * @override
+     */
+    async fetchData() {
+        const remaining = this.totalToFetch - this.data.length;
+        this.limit = Math.min(this.chunkSize, remaining);
+        // show loading effect
+        this.content.classList.add("d-none");
+        this.holder.classList.remove("d-none");
+        await super.fetchData();
+        const fragments = this.data;
+        this.fetchedData.push(...fragments);
+        this.data = this.fetchedData;
+        this.offset += this.limit;
+        this.hasMore = this.data.length < this.totalToFetch && fragments.length === this.limit;
+        if (this.carouselEl) {
+            this.appendSlides();
+        }
+        // hide loading effect
+        this.holder.classList.add("d-none");
+        this.content.classList.remove("d-none");
+    }
+
+    /**
+     * Append new slides to the current carouselEl.
+     */
+    appendSlides() {
```

Comment:
Can't we make solution more generic so that after fetching extra data we just need to call renderContent or render method and it will do the job and it will work in carousel, blogs all places.